GNOME #436811; Prepend the "C" locale instead of "en_US" as "C" is always
available while "en_US" might not be.

This is pre 2.6.0.8-1.

Index: gdm-2.18.1/gui/gdmlanguages.c
===================================================================
--- gdm-2.18.1.orig/gui/gdmlanguages.c	2007-04-09 07:15:10.000000000 +0200
+++ gdm-2.18.1/gui/gdmlanguages.c	2007-05-02 11:08:20.000000000 +0200
@@ -296,6 +296,7 @@
 gdm_lang_init (void)
 {
 	int i;
+
 	if (lang_names != NULL)
 		return;
 
@@ -524,7 +525,6 @@
 	char curline[256];
 	GList *langs = NULL;
 	GHashTable *dupcheck;
-	gboolean got_english = FALSE;
 	Language *language;
 	gboolean clean;
 	char *curlocale;
@@ -616,21 +616,15 @@
 		g_hash_table_insert (dupcheck, g_strdup (lang),
 				     GINT_TO_POINTER (1));
 
-		/* if we have an english locale */
-		if (strncmp (lang, "en_", 3) == 0 ||
-		    strcmp (lang, "C") == 0)
-			got_english = TRUE;
-
 		g_strfreev (lang_list);
 	}
 
 	g_hash_table_foreach (dupcheck, (GHFunc) g_free, NULL);
 	g_hash_table_destroy (dupcheck);
 
-	/* If we haven't found any english locale, add american
-	 * english as that's as much of a fallback as we can get */
-	if ( ! got_english)
-		langs = g_list_prepend (langs, g_strdup ("en_US"));
+	/* Ensure there is at least one English locale -- the one available
+	 * on all systems, "C" */
+	langs = g_list_prepend (langs, g_strdup ("C"));
 
 	curlocale = setlocale (LC_MESSAGES, NULL);
 	if (curlocale != NULL &&
